cfgov/core/jinja2/core/docs/streamfield-blocks.md

```diff
@@ -0,0 +1,94 @@
+<!--
+
+Do not edit this file manually!
+
+The contents of this file are generated automatically as part of this project's
+documentation build process.
+
+StreamField block and model information is extracted from the Python source
+code. To modify this file's layout, see the Markdown template file at:
+
+cfgov/core/jinja2/core/docs/streamfield-blocks.md.
+
+To regenerate this file, run the following Django management command from
+the project root when inside of a Python environment:
+
+cfgov/manage.py generate_streamfield_docs -o docs/streamfield-blocks.md
+
+These instructions are duplicated at
+[Updating the documentation](../development-tips/#tip-updating-the-documentation).
+
+-->
+# StreamField blocks
+
+This project makes use of
+[Wagtail StreamFields](http://docs.wagtail.io/en/v1.13.4/topics/streamfield.html)
+to allow page editors to enter freeform page content.
+
+## Table of contents
+
+[TOC]
+
+## StreamField block types by module
+
+This project includes numerous
+[custom StreamField block types](http://docs.wagtail.io/en/v1.13.4/topics/streamfield.html#structural-block-types)
+that allow for consistent use of design patterns across the website. Many of
+these blocks implement patterns described by the
+[CFPB Design Manual](https://cfpb.github.io/design-manual/).
+
+{% set moduleblocks = blocks|sort(attribute='module')|groupby('module') -%}
+{% for module, blocks in moduleblocks %}
+### `{{ module }}`
+
+{% for block in blocks|sort(attribute='name') -%}
+#### {{ block.name }}
+
+- Module: `{{ block.fullname }}`
+- Template: `{{ block.template|default('none') }}`
+{% if block.fields %}
+- Available in:
+{% for field in block.fields|sort %}
+    - `{{ field }}`
+{%- endfor %}
+{%- endif %}
+
+{% if block.docstring %}
+{{ block.docstring }}
+{%- endif %}
+{%- endfor %}
+
+{%- endfor %}
+
+{% macro showmodels(models) -%}
+{% for model in models|sort(attribute='name') -%}
+#### {{ model.name }}
+
+- Module: `{{ model.fullname }}`
+- StreamFields:
+{% for field in model.streamfields %}
+    - `{{ field.name }}`{% if field.required %} (required){% endif %}
+{%- endfor %}
+
+{% if model.docstring -%}
+{{ model.docstring }}
```

Comment:
It's kinda weird that the model.docstrings just return a list of all available stuff in that model if a real docstring isn't found. I'm not sure it adds much value; maybe if we could ignore (or list once at the top) all the stuff inherited from `Page`/`CFGOVPage`?

Reply:
I also worry slightly that docstrings are (traditionally) written as Restructured Text-ish because of `pydoc` (although that might just me showing my age) and it's included directly as Markdown. It might not be a bad idea to document that the docstrings will be rendered as Markdown.

Reply:
I agree with @Scotchester though that this is weird in both formatting and along side the models that do have docstrings:

![image](https://user-images.githubusercontent.com/10562538/50161114-dbfc0a00-02a8-11e9-92bf-d83c420a9fce.png)

It took me a while to figure out where that was coming from to comment on it.